ui/components/multichain/badge-status/badge-status.js

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { useSelector } from 'react-redux';
+import {
+  AlignItems,
+  BackgroundColor,
+  BorderColor,
+  BorderRadius,
+  Color,
+  Display,
+  JustifyContent,
+  Size,
+} from '../../../helpers/constants/design-system';
+import {
+  AvatarAccount,
+  AvatarAccountVariant,
+  BadgeWrapper,
+  Box,
+} from '../../component-library';
+import { getUseBlockie } from '../../../selectors';
+import Tooltip from '../../ui/tooltip';
+
+export const BadgeStatus = ({
+  className = '',
+  badgeBackgroundColor = Color.borderMuted,
+  badgeBorderColor = BackgroundColor.backgroundDefault,
+  address,
+  isConnectedAndNotActive = false,
+  text,
+}) => {
+  const useBlockie = useSelector(getUseBlockie);
+
+  return (
+    <Box
+      className={classNames('multichain-badge-status', className)}
+      data-testid="multichain-badge-status"
+      as="button"
+      display={Display.Flex}
+      alignItems={AlignItems.center}
+      justifyContent={JustifyContent.center}
+      backgroundColor={BackgroundColor.backgroundDefault}
+    >
+      <Tooltip
+        title={text}
+        data-testid="multichain-badge-status__tooltip"
+        position="bottom"
+      >
+        <BadgeWrapper
+          positionObj={
+            isConnectedAndNotActive
+              ? { bottom: 2, right: 5, zIndex: 1 }
+              : { bottom: -1, right: 2, zIndex: 1 }
+          }
+          badge={
+            <Box
+              className={classNames('multichain-badge-status__badge', {
+                'not-connected': isConnectedAndNotActive,
```

Comment:
This classname feels too generic.  Better would be `multichain-badge-status__badge--not-connected`

Reply:
Updated https://github.com/MetaMask/metamask-extension/pull/22972/commits/f1de41abfbe320e59a8815c56b5f6c951deb19d0